src/utils/useExtent.ts

```diff
@@ -0,0 +1,23 @@
+import { useMemo } from 'react';
+
+/**
+ * This hook returns the minimum and maximum values for the given array.
+ * Optionally a predicate can be passed in to extract the number values from the given array.
+ *
+ * @param collection the given array
+ * @param predicate the optional predicate
+ */
+export default function useExtent(
+  collection: any[],
+  predicate?: (item: any) => number
```

Comment:
This might be a bit more correct.
```suggestion
  predicate?: (item: Record<string, unknown>) => number
```

Reply:
Or if arrays are supported as well.

```ts
Record<string, unknown> | Array<unknown>
```